packages/gatsby-react-router-scroll/src/ScrollBehaviorContext.ts

```diff
@@ -14,25 +17,44 @@ const childContextTypes = {
   scrollBehavior: PropTypes.object.isRequired,
 }
 
-class ScrollContext extends React.Component {
-  constructor(props, context) {
+interface IProps {
+  shouldUpdateScroll: ShouldUpdateScroll<IContext>
+  children: React.ReactElement
+  location: LocationBase
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface IContext {}
```

Comment:
instead of using an empty context for a variable we aren't using, we can actually just drop the context from the constructor entirely.

e.g.

```
constructor(props: IProps) {
  super(props)
  ...
}